src/datasources/asset-calibration/constants.ts

```diff
@@ -0,0 +1,84 @@
+import { ComputedDataFields, QueryBuilderOperations } from "core/query-builder.constants";
+import { AssetTypeOptions, BusTypeOptions, QBField } from "./types";
+
+export const AssetCalibrationFields: Record<string, QBField> = {
+    LOCATION: {
+        label: 'Location',
+        dataField: ComputedDataFields.Location,
```

Comment:
Actually, I don't think I agree with the approach we've taken here for the "computed" fields. If I'm understanding correctly, what we're trying to do is: 
when a user inputs `Location = "foo"` in the query builder, it expands to `Location.MinionId = "foo" || Location.PhysicalLocation = "foo"`. But that expanded filter gets saved in query model, so we have to call `setQueryComputedFields()` to remove the extra filter so the query builder can parse it back to what the user expects.

Why don't we just save `Location = "foo"` in the filter, and then expand it when we make the request?

Reply:
I came up with another approach that takes into consideration your suggestion. I created a function `transformComputedFieldsQuery` that takes as a parameter the query and an object that holds the template for the transformation. This will be called before sending the query.